Sources/PackageModel/Manifest/TargetDescription.swift

```diff
@@ -110,6 +110,17 @@ public struct TargetDescription: Equatable, Codable {
     /// Represents the declared capability of a package plugin.
     public enum PluginCapability: Equatable {
         case buildTool
+        case command(intent: PluginCommandIntent, permissions: [PluginPermission])
+    }
+    
+    public enum PluginCommandIntent: Equatable, Codable {
+        case documentationGeneration
+        case sourceCodeFormatting
+        case custom(verb: String, description: String)
+    }
```

Comment:
I wonder if intents should have default permissions?

Reply:
Yes, I've been thinking the same, although because it deals with security it seemed reasonable to be very clear about what is going on.  It's also possible that not all source code formatting plugins need source code writability, for example, if they for example generate fixits instead of actually change the source code.  But perhaps that would be a different category.